src/main/java/org/passay/AbstractDictionaryRule.java

```diff
@@ -76,18 +76,20 @@ public RuleResult validate(final PasswordData passwordData)
   {
     final RuleResult result = new RuleResult(true);
     String text = passwordData.getPassword();
-    String matchingWord = doWordSearch(text);
-    if (matchingWord != null) {
-      result.setValid(false);
-      result.getDetails().add(new RuleResultDetail(ERROR_CODE, createRuleResultDetailParameters(matchingWord)));
-    }
-    if (matchBackwards && text.length() > 1) {
-      text = new StringBuilder(passwordData.getPassword()).reverse().toString();
-      matchingWord = doWordSearch(text);
+    if (text.length() > 1) {
+      String matchingWord = doWordSearch(text);
       if (matchingWord != null) {
         result.setValid(false);
-        result.getDetails().add(
-          new RuleResultDetail(ERROR_CODE_REVERSED, createRuleResultDetailParameters(matchingWord)));
+        result.getDetails().add(new RuleResultDetail(ERROR_CODE, createRuleResultDetailParameters(matchingWord)));
+      }
+      if (matchBackwards) {
+        text = new StringBuilder(passwordData.getPassword()).reverse().toString();
+        matchingWord = doWordSearch(text);
+        if (matchingWord != null) {
+          result.setValid(false);
+          result.getDetails().add(
+            new RuleResultDetail(ERROR_CODE_REVERSED, createRuleResultDetailParameters(matchingWord)));
+        }
```

Comment:
Ah, that can be left out as it is indeed unrelated to the issue.  The purpose was that since we are checking for empty string on the backwards search, we should do so on the forward search as well.